File: client/src/application/assignments/use_cases/get_assignments.use_case.js
class GetAssignmentsUseCase {
  constructor(assignmentRepository) {
    this.assignmentRepository = assignmentRepository;
  }

  async execute(req) {
    return await this.assignmentRepository.getAssignments(
      req,
    );
  }
}

module.exports = GetAssignmentsUseCase;

File: server/apps/assignments/application/use_cases/get_assignments_use_case.py
class GetAssignmentsUseCase:

    def __init__(self, assignment_repository):
        self.assignment_repository = assignment_repository

    def execute(
        self,
        university_id: str,
        teacher_email: str,
    ):
        return self.assignment_repository.get_assignments(
            university_id=university_id,
            teacher_email=teacher_email,
        )